src/main/java/interface_adapters/rate/RateState.java

```diff
@@ -35,19 +35,19 @@ public void setRating(int rating) {
         this.rating = rating;
     }
 
-    public void setRatingError(String errorMessage) {
-        this.ratingerror = errorMessage;
-    }
-
     public String getRatingError() {
         return ratingerror;
     }
 
-    public void setCommentError(String errorMessage) {
-        this.commentError = commentError;
+    public void setRatingError(String errorMessage) {
+        this.ratingerror = errorMessage;
     }
 
```

Comment:
Nit: This getter goes up too. They should be in the same order as their attributes are initialized.

ex:
```
comment
username
ratingError

get comment
get username
get ratingError
```